kn/leden/mongo.py

```diff
@@ -33,9 +38,20 @@ def delete(self):
     def save(self, update_fields=NotImplemented):
         if self._parent is None:
             if '_id' in self._data:
-                self._collection.update({
-                    '_id': self._data['_id']}, self._data)
+                if self._versioned:
+                    self._data['_version'] += 1
+                    result = self._collection.update({
+                        '_id': self._data['_id'],
+                        '_version': self._data['_version']-1}, self._data, w=1)
+                    if result['n'] < 1:
+                        raise MongoWrapperException(
+                            'Could not save - version mismatch?')
```

Comment:
Ik zou hier een aparte exceptionclass van maken: `RaceCondition`.  Dan is er mee te programmeren op deze manier:

```
while True:
    bla = Es.by_name('bla')
    stuff(bla)
    try:
        bla.save()
    except RaceCondition:
        continue # try again
    break
```